src/Microsoft.OData.Core/MultipartMixed/ODataMultipartMixedBatchReader.cs

```diff
@@ -85,7 +92,14 @@ protected override ODataBatchOperationRequestMessage CreateOperationRequestMessa
 
                     if (this.currentContentId == null)
                     {
-                        throw new ODataException(Strings.ODataBatchOperationHeaderDictionary_KeyNotFound(ODataConstants.ContentIdHeader));
+                        if (requireContentId)
+                        {
+                            throw new ODataException(Strings.ODataBatchOperationHeaderDictionary_KeyNotFound(ODataConstants.ContentIdHeader));
+                        }
+                        else
+                        {
+                            this.currentContentId = Guid.NewGuid().ToString();

```

Comment:
Is it better to change the type of 'CurrentContentId' to System.Guid?

Reply:
currentContentId is an arbitrary value of user-provided header. I assign it to guid just for specific case